### PR TITLE
Add WaitAny and WaitAll tests for duplicate handle cases

### PR DESCRIPTION
**Pull request by kouvel**
Related to dotnet/coreclr#630

### CODE REVIEW

**Comment by kouvel** [comment]
CC @sergiy-k @adityamandaleeka @stephentoub

File: src/System.Runtime/tests/System/Threading/WaitHandle.cs [hunk 1]
```diff
@@ -62,6 +72,25 @@ public static void WaitAll()
...
+        ManualResetEvent[] wh = new ManualResetEvent[2];
+        wh[0] = new ManualResetEvent(true);
+        wh[1] = wh[0];
+
+        // Assert.Throws() requires checking for the exact type of exception. This test is only concerned with checking for
+        // ArgumentException or any derived types, as recommended in the docs in this particular case for some platforms.
+        try
+        {
+            WaitHandle.WaitAll(wh);
+            Assert.True(false, "Expected an exception, got no exception.");
```
Comment (Maxwe11) [inline src/System.Runtime/tests/System/Threading/WaitHandle.cs]: `Assert.ThrowsAny`

File: src/System.Runtime/tests/System/Threading/WaitHandle.cs [hunk 2]
```diff
@@ -43,6 +43,16 @@ public static void WaitAny()
     }
 
     [Fact]
+    public static void WaitAnySameHandles()
+    {
+        ManualResetEvent[] wh = new ManualResetEvent[2];
+        wh[0] = new ManualResetEvent(true);
+        wh[1] = wh[0];
+
+        Assert.Equal(WaitHandle.WaitAny(wh), 0);
```
Comment (Maxwe11) [inline src/System.Runtime/tests/System/Threading/WaitHandle.cs]: expected argument should be passed first

    Assert.Equal(0, WaitHandle.WaitAny(wh));

Comment (kouvel) [inline src/System.Runtime/tests/System/Threading/WaitHandle.cs]: Thanks for those pointers, done

**Comment by stephentoub** [comment]
LGTM

**Comment by stephentoub** [comment]
@kouvel, this fails with the error:
```C#
WaitHandleTests.WaitAllSameHandles [FAIL]
18:49:18       Assert.Throws() Failure
18:49:18       Expected: typeof(System.ArgumentException)
18:49:18       Actual:   (No exception was thrown)
18:49:18       Stack Trace:
18:49:18             at WaitHandleTests.WaitAllSameHandles()
```
I assume this needs https://github.com/dotnet/coreclr/pull/1554 to be merged before this will pass?

**Comment by kouvel** [comment]
Yea this has to be merged in after that one, I've tested with both sets of changes and it works

**Comment by stephentoub** [comment]
@dotnet-bot test this please

**Comment by stephentoub** [comment]
(CoreCLR is currently failing some PAL tests on Linux, such that a new libcoreclr that has the changes from https://github.com/dotnet/coreclr/pull/1554 isn't being picked up by corefx CI yet.)

**Comment by kouvel** [comment]
Squashed into one commit

**Pull request merged**